other/js/auth_plugin_examples.js

```diff
@@ -0,0 +1,90 @@
+/*
+ * An auth plugin must be a function which returns a function conforing to the
+ * requirements of the `verifyClients` option on ws.WebSocket.Server.
+ *
+ * See: https://github.com/websockets/ws/blob/master/doc/ws.md
+ *
+ * If websockify is provided with an --auth-source argument, this will be
+ * passed to the auth plugin as its first argument.
+ *
+ */
+
+const querystring = require('querystring');
+const fs = require('fs');
+
+function urlTokenMatch(url, token, verbose=false) {
+    /**
+     * Parse the url path, extract the `token` querystring value, and check if
+     * it matches the token argument. If verbose is set to true, log messages
+     * are enabled.
+     *
```

Comment:
The "token" query parameter already has a different meaning in the main python implementation which is used for selecting among multiple target hosts: https://github.com/novnc/websockify/wiki/Token-based-target-selection. The python implementation also supports basic authentication via the Authorization header. I think it would be better to have an example of basic auth as that would probably be more generally applicable. But certainly it shouldn't be using "token" for authentication.

Reply:
Replaced token based auth examples with origin-based examples (these being easier to test out using NoVNC than basic auth): https://github.com/novnc/websockify/pull/285/commits/a543c359438b2368010cff352d07148e1136295e